File: go/rna-transcription/rna_transcription.go
package strand

import (
	"strings"
)

var conversions = map[rune]rune{
	'C': 'G',
	'G': 'C',
	'A': 'U',
	'T': 'A',
}

func convert(nucleotide rune) rune {
	value, present := conversions[nucleotide]
	if present {
		return value
	}

	panic("invalid nucleotide")
}

func ToRna(strand string) string {
	return strings.Map(convert, strand)
}

File: erlang/rna-transcription/rna_transcription.erl
-module(rna_transcription).
-export([to_rna/1]).

to_rna(Seq) ->
    lists:map(fun convert/1, Seq).

convert($G) -> $C;
convert($C) -> $G;
convert($T) -> $A;
convert($A) -> $U.
